### PR TITLE
RFC-0024-Process-for-RFC-Shepherding

### PR DESCRIPTION
**Pull request by suraj813**
This document proposes a controlled and consistent RFC shepherding process from proposal to implementation, along with a standard RFC template for authors to follow.  

[Rendered](https://github.com/suraj813/rfcs/blob/rfc-process/RFC-0024-rfc-process.md)

### CODE REVIEW

**Review by rgommers — APPROVED** [review]
This looks great to me, thanks @suraj813. Only one comment from me.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,65 @@
...
+ 
+- Fork the rfcs repo: https://github.com/pytorch/rfcs
+- Copy `RFC-0000-template.md` to `RFC-00xx-my-feature.md`, or write your own open-ended proposal. Put care into the details.
+- Submit a pull request titled `RFC-OOxx-my-feature`. Assign the `commenting` label on the PR to open it for discussions. 
+- Build consensus for your proposal, integrate feedback and revise it as needed, and summarize the outcome of the discussion via a [resolution template](https://github.com/pytorch/rfcs/blob/rfc-process/RFC-0000-template.md#resolution).
+    - If the RFC is idle here (no activity for 2 weeks), assign the label `stalled` to the PR.
+- Once the discussion has settled, assign a new label based on the level of support:
+    - `accepted` if a decision has been made in the RFC
+    - `draft` if the author needs to rework the RFC’s proposal
+    - `postponed `if there are no plans to move ahead with the current RFC’s proposal
```
[the remark this answers is not in the document]
Comment (kumpera) [reply README.md:30]: If `accepted` follows the resolution template, there's room for a rejected outcome, just the naming that's odd. Right?

[the remark this answers is not in the document]
Comment (suraj813) [reply README.md:30]: You're right, I meant for `postponed`  to imply that we're not thinking about this right now, but there is at least a small chance we might revisit it, but the meaning can be ambiguous. I suppose renaming it to `shelved`, and adding a new status of `rejected` might disambiguate?

[the remark this answers is not in the document]
Comment (suraj813) [reply README.md:30]: Renamed it to `shelved` and added a clarification on the implication:
```
- `shelved` if there are no plans to move ahead with the current RFC’s proposal. We want neither to think about evaluating the proposal nor about implementing the described feature until some time in the future.
```

**Comment by ezyang** [comment]
This looks good. One thing that I have noticed is that as implementation proceeds, some details of the RFC often need to get tweaked. And of course, when the RFC is closed, we may still continue evolving the feature as time goes on. I've noticed that I have never had time to revisit old RFCs to update their information, but I really wish that we could keep design documents alive and updated as we go along. Maybe that should be some other mechanism.

**Comment by ezyang** [comment]
Another thought: while an RFC is in draft phase, the authors may find it easier to compose the RFC in a WYSIWYG editor (e.g., Google Docs). I think we should officially support this workflow. The idea is that when you are not in general commenting (but maybe you are sharing the doc with a few close collaborators), it doesn't really matter where you put it, and then when the RFC graduates into general commenting, you should put it into the GitHub format. Ideally, the GDoc is public even in the drafting phase; fortunately, @dzhulgakov has a good set of instructions for FB authors on how to work around the "default private" settings on our corp gdocs.

**Comment by suraj813** [comment]
> authors may find it easier to compose the RFC in a WYSIWYG editor (e.g., Google Docs). I think we should officially support this workflow.

I updated the workflow instructions to be:
```
- Submit a pull request titled `RFC-00xx-my-feature`. 
    - Assign the `draft` label while composing the RFC. You may find it easier to use a WYSIWYG editor (like Google Docs) when working with a few close collaborators; feel free to use whatever platform you like. Ideally this document is publicly visible and is linked to from the PR.
    - When opening the RFC for general discussion, copy your draft into the `RFC-00xx-my-feature.md` file on the PR and assign the `commenting` label.
```

**Review by albanD — APPROVED** [review]
Let's go!

**Pull request merged**